allofplos/plos_regex.py

```diff
@@ -22,6 +22,10 @@
 full_doi_regex_match = re.compile(regex_match_prefix+regex_body_match)
 full_doi_regex_search = re.compile(r"10\.1371/journal\.p[a-zA-Z]{3}\.[\d]{7}"
                                    "|10\.1371/annotation/[a-zA-Z0-9]{8}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{12}")
+partial_doi_regex_search = re.compile(r"p[a-zA-Z]{3}\.[\d]{7}"
+                                      "|annotation/[a-zA-Z0-9]{8}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{12}")
+partial_doi_regex_match = re.compile(r"^p[a-zA-Z]{3}\.[\d]{7}$"
```

Comment:
It looks like this is just the same string with `^`+your_str+`$`, am I correct? and the last part of it is almost the same as the last part of the full_doi_regex_search? 

Why not make this based on a single string that you then combine as needed to make the search and match?

 I'm not sure you need to define search vs. match separately anyway (in this case at least). I ask below… but it'd be a lot easier to know whether that is the case if you tested the regexes directly.